### ISSUE TITLE
SIGILL in census benchmark in scikit-learn code when running it with HDK

### ISSUE THREAD

**Issue by gshimansky** [intel-ai/hdk#167]
I run census benchmark from https://github.com/gshimansky/data-science-processing-workload repo. Both on randomly generated dataset and using original ipums_education2income_1970-2010.csv file I am getting the same behavior. When I run this benchmark with HDK engine the code starts to execute ML part and gets SIGILL somewhere in daal4py code. Here is a native backtrace and offending instruction:
```
Thread 192 "python" received signal SIGILL, Illegal instruction.
[Switching to Thread 0x7ff8aa4fc700 (LWP 83772)]
0x00007ffd4261309c in poly_mul_kar ()
   from /home/gregory/work/miniconda3/envs/modin-0.18.0/lib/python3.10/site-packages/daal4py/../../../libonedal_core.so
(gdb) bt
#0  0x00007ffd4261309c in poly_mul_kar ()
   from /home/gregory/work/miniconda3/envs/modin-0.18.0/lib/python3.10/site-packages/daal4py/../../../libonedal_core.so
#1  0x00007ffd42618867 in fpk_vsl_sub_kernel_z0_vslPolyRem ()
   from /home/gregory/work/miniconda3/envs/modin-0.18.0/lib/python3.10/site-packages/daal4py/../../../libonedal_core.so
#2  0x00007ffd424ad63c in fpk_vsl_sub_kernel_z0_vsl_SkipAheadSWin ()
   from /home/gregory/work/miniconda3/envs/modin-0.18.0/lib/python3.10/site-packages/daal4py/../../../libonedal_core.so
#3  0x00007ffd424797c5 in fpk_vsl_sub_kernel_z0_BRngMT19937InitStream ()
   from /home/gregory/work/miniconda3/envs/modin-0.18.0/lib/python3.10/site-packages/daal4py/../../../libonedal_core.so
#4  0x00007ffd424180e6 in fpk_vsl_sub_kernel_z0_vslSkipAheadStream ()
   from /home/gregory/work/miniconda3/envs/modin-0.18.0/lib/python3.10/site-packages/daal4py/../../../libonedal_core.so
#5  0x00007ffd3d977d09 in void daal::threader_func<daal::data_management::internal::generateShuffledIndicesImpl<int, (daal::CpuType)6>(daal::services::interface1::SharedPtr<daal::data_management::interface1::NumericTable> const&, daal::services::interface1::SharedPtr<daal::data_management::interface1::NumericTable> const&)::{lambda(unsigned long)#1}>(int, void const*) ()
   from /home/gregory/work/miniconda3/envs/modin-0.18.0/lib/python3.10/site-packages/daal4py/../../../libonedal_core.so
#6  0x00007ffd3971b104 in tbb::detail::d1::start_for<tbb::detail::d1::blocked_range<int>, _daal_threader_for::{lambda(tbb::detail::d1::blocked_range<int>)#1}, tbb::detail::d1::auto_partitioner const>::execute(tbb::detail::d1::execution_data&) ()
   from /home/gregory/work/miniconda3/envs/modin-0.18.0/lib/python3.10/site-packages/daal4py/../../../libonedal_thread.so
#7  0x00007ffd395a5dda in tbb::detail::r1::task_dispatcher::local_wait_for_all<false, tbb::detail::r1::outermost_worker_waiter> (t=0x7ffb9be7d800, 
    waiter=..., this=0x7ff8ab10ec80)
    at /home/conda/feedstock_root/build_artifacts/tbb_1670332502460/_build_env/x86_64-conda-linux-gnu/include/c++/10.4.0/atomic:261
#8  tbb::detail::r1::task_dispatcher::local_wait_for_all<tbb::detail::r1::outermost_worker_waiter> (t=0x0, waiter=..., this=0x7ff8ab10ec80)
    at /usr/local/src/conda/tbb-2021.7.0/src/tbb/task_dispatcher.h:458
#9  tbb::detail::r1::arena::process (this=<optimized out>, tls=...) at /usr/local/src/conda/tbb-2021.7.0/src/tbb/arena.cpp:137
#10 0x00007ffd395b29fb in tbb::detail::r1::market::process (j=..., this=0x7ff8ab3ef080) at /usr/local/src/conda/tbb-2021.7.0/src/tbb/market.cpp:593
#11 tbb::detail::r1::rml::private_worker::run (this=<optimized out>) at /usr/local/src/conda/tbb-2021.7.0/src/tbb/private_server.cpp:271
#12 tbb::detail::r1::rml::private_worker::thread_routine (arg=0x7ff8ab105080) at /usr/local/src/conda/tbb-2021.7.0/src/tbb/private_server.cpp:221
#13 0x00007ffff7f8d609 in start_thread (arg=<optimized out>) at pthread_create.c:477
#14 0x00007ffff7d4e133 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95
(gdb) i r rip
rip            0x7ffd4261309c      0x7ffd4261309c <poly_mul_kar+6444>
(gdb) x/1i 0x7ffd4261309c
=> 0x7ffd4261309c <poly_mul_kar+6444>:  vpclmullqlqdq %xmm5,%xmm4,%xmm23
(gdb) 
   0x7ffd426130a3 <poly_mul_kar+6451>:  vmovq  %rdx,%xmm16
```
Processor is Intel(R) Xeon(R) Gold 6242R with support for multiple AVX extensions including AVX512 subsets avx512f avx512dq  avx512cd avx512bw avx512vl avx512_vnni.

**Comment by Garra1980** [comment]
fail is in scikit-learn library as far as I can see - can we probably ask Intel ML folks to check what exactly happens there?

**Comment by gshimansky** [comment]
Yes the crash happens inside of daal4py but it happens there only when HDK is involved.

**Comment by gshimansky** [comment]
Apparently this bug is not related to HDK. The same SIGILL happens when modin backend is Ray.

**Comment by gshimansky** [comment]
The problem is fixed in oneDAL 2023.0.1. See https://github.com/intel/scikit-learn-intelex/issues/1167#issuecomment-1404032064 for details.
